docs/DeveloperGuide.md

```diff
@@ -262,31 +262,272 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `Pawbook` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case: UC01 - Add a dog profile**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User request to add a dog profile to the list.
+2.  Pawbook adds the dog.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. Dog name is not specified.
 
-  Use case ends.
+    * 1a1. Pawbook shows an error message.
+    * 1a2. User adds dog name.
 
-* 3a. The given index is invalid.
+      Use case resumes at step 2.
+
+* 1b. Dog breed is not specified.
+
+    * 1b1. Pawbook shows an error message.
+    * 1b2. User adds dog breed.
+
+      Use case resumes at step 2.
+
+* 1c. Owner ID is not specified.
+
+    * 1c1. Pawbook shows an error message.
+    * 1c2. User adds owner ID.
+
+      Use case resumes at step 2.
+
+* 1d. Tag is not specified.
+
+    * 1d1. Pawbook shows an error message.
+    * 1d2. User adds tag.
+
+      Use case resumes at step 2.
```

Comment:
I think this is the same issue as what you have raised about separation of the add/delete of dog/owner, in this case shall we just keep things simple and simply call it "missing required information"? To maintain clarify we can put a note at the bottom saying what field are required for dogs and owners. Doing it this way reduces clutter since we can combine more things together, makes any future edits less likely to miss out any detail.